src/mame/video/tecmosys.cpp

```diff
@@ -264,53 +204,62 @@ void tecmosys_state::do_final_mix(bitmap_rgb32 &bitmap)
 
 				dstptr[x] = b | (g<<8) | (r<<16);
 			}
-			else if (pri2 >= pri)
+			else if ((pri2 >= pri) && ((mask != 0x3f00) || (mask != 0)))
```

Comment:
This doesn't make sense - the condition `((mask != 0x3f00) || (mask != 0))` cannot be false, as `mask` cannot be equal to both 0x3f00 and 0 at the same time.